src/metabase/pulse/render/body.clj

```diff
@@ -550,27 +550,31 @@
   [col-a col-b]
   (let [[min-a min-b]    (map #(get-in % [:fingerprint :type :type/Number :min]) [col-a col-b])
         [max-a max-b]    (map #(get-in % [:fingerprint :type :type/Number :max]) [col-a col-b])
-        non-overlapping? (or (and (< max-a min-b) (< max-a max-b))
-                             (and (> min-a min-b) (> min-a max-b)))]
-   (if non-overlapping?
-     0
-     (let [[a b c d]     (sort [min-a min-b max-a max-b])
-           max-width     (- d a)
-           overlap-width (- c b)]
-       (/ overlap-width max-width)))))
+        valid-ranges?    (and min-a min-b max-a max-b)
+        overlapping-and-valid? (and valid-ranges?
+                                    (or (<= min-a min-b max-a)
+                                        (<= min-a max-b max-a)))]
+    (if
+      overlapping-and-valid?
+      (let [[a b c d]     (sort [min-a min-b max-a max-b])
+            max-width     (- d a)
+            overlap-width (- c b)]
+        (/ overlap-width max-width))
+      0)))
 
 (defn- group-axes
   [cols-meta group-threshold]
-  (let [cols-by-type (group-by (juxt :base_type :effective_type :semantic_type) cols-meta)]
-    (when (not= (count cols-by-type) (count cols-meta))
-      (let [num?               (fn [[[col-type]]] (isa? col-type :type/Number))
-            {num-cols   true
-             other-cols false} (-> (group-by num? cols-by-type)
-                                   (update-vals #(mapcat second %)))
-            first-axis         (first num-cols)
-            grouped-num-cols   (-> (group-by #(> (overlap first-axis %) group-threshold) num-cols)
+  (when-let [groupable-cols (->> cols-meta
+                                 (filter #(isa? (:base_type %) :type/Number)) ;; for now we only try grouping number cols
+                                 (remove (comp nil? :fingerprint)) ;; we can't group if there is no fingerprint
+                                 seq)]
+    (let [cols-by-type  (group-by (juxt :base_type :effective_type :semantic_type) groupable-cols)
+          some-grouped? (> (last (sort (map #(count (second %)) cols-by-type))) 1)]
```

Comment:
NBD, but I think `(last (sort ...))` is `(apply max ...)`

Reply:
I changed it for such a case: 

```clojure
(apply max [])   ;; -> wrong number of args passed to max
(last (sort [])) ;; -> nil
```

Reply:
Ohh, good point